### DISCUSSION TITLE
Toolbox brick permission

### DISCUSSION THREAD

**Discussion by federicobernardin** [dachcom-digital/pimcore-toolbox#235] [Q&A]
Hello everyone,

I'm currently using this extension bundle, but I'm struggling to find clear information on how to enable the usage of the bricks toolbox for non-admin users in Pimcore.

When a non-admin user tries to edit a brick on a page, the "Edit" feature is missing, and the brick appears only in "view" mode.

Has anyone encountered this issue before? Is there a specific configuration or permission setting required to enable brick editing for non-admin users?

Any help or guidance would be greatly appreciated!

Thank you in advance!

**Comment by mzweifel** [comment]
Hi,

Please check the permissions for the bricks in the users or roles.

There is a maintenance task that updates the brick permissions.
https://github.com/dachcom-digital/pimcore-toolbox/blob/master/src/Maintenance/AreaPermissionTask.php

**Reply by federicobernardin** [reply]
Hi, thank you for your sudden answer. In the user or or roles permissions I didn't find any toolbox permissions about the bricks.
The link you wrote is about a task. Should I run that task to see the permissions in user or roles?

**Reply by solverat** [reply]
@federicobernardin nope, this task runs automatically, if you've enabled the pimcore maintenance task via cron. this task "only" fetches (newly) available areas/bricks and will add them to the permission table. It will NOT assign the permission to any user but only creates the availability of the permission.

So basically you've to set the permissions on user or role layer (role layer is recommended) so your users can use the bricks.

**Reply by federicobernardin** [reply]
Hi @solverat, we have just run the cron, but we didn't find any checkbox related to bricks in role permissions (similar to other bundles I supposed). Are there any configuration not described in wiki to enable permissions creation?
